src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,208 @@
+            throw new DataProcessingException(
+                    "Couldn't get a list of cars by driver with id " + driverId, e);
+        }
+        cars.forEach(this::insertDriversToCar);
+        return cars;
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("car_id", Long.class);
+        String model = resultSet.getString("model");
+        Manufacturer manufacturer = new Manufacturer(
+                resultSet.getObject("manufacturer_id", Long.class),
+                resultSet.getString("name"),
+                resultSet.getString("country"));
+        return new Car(id, model, manufacturer);
+    }
+
+    private List<Driver> getDriversByCar(Long id) {
+        List<Driver> drivers = new ArrayList<>();
+        String query = "SELECT id, name, license_number "
+                + "FROM cars_drivers cd "
+                + "JOIN drivers d "
+                + "ON cd.driver_id = d.id "
+                + "WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Driver driver = new Driver(
+                        resultSet.getObject("id", Long.class),
+                        resultSet.getString("name"),
+                        resultSet.getString("license_number"));
+                drivers.add(driver);
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get drivers by car id " + id, e);
+        }
+    }
+
+    private void deleteDriversFromCar(Long id) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete driver from"
+                    + " carsDB with id " + id, e);
+        }
+    }
+
+    private void insertDriversToCar(Car car) {
+        String insertRequest = "INSERT INTO cars_drivers (driver_id, car_id) "
+                + "VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(insertRequest)) {
+            statement.setLong(2, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(1, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't insert driver to DB. Car: " + car, e);
```

Comment:
Why do you use here RuntimeException?

Reply:
Agree, let's use `DataProcessingException`.